Feature/hee 262 hee 272 search filter updates

Raq-HEE opened the pull request:
For HEE-272 I've added 'hee:event', 'hee:caseStudy', 'hee:listingPage' to include in the search results page and some test content.

For HEE-262 I wasn't sure whether there'd be a requirement to reintroduce this functionality with more relevant filter types, so have commented this out in the ftl template for now.

smanivasag commented on repository-data/application/src/main/resources/hcm-config/namespaces/hee/event.yaml:
```diff
@@ -60,6 +60,57 @@ definitions:
             hipposysedit:primary: false
             hipposysedit:type: String
             hipposysedit:validators: [required]
+        /hipposysedit:nodetype[2]:
```
Looks like a duplicate node - could we remove if its redundant ?

Raq-HEE replied:
Removed.

smanivasag commented on repository-data/application/src/main/resources/hcm-config/namespaces/hee/event.yaml:
```diff
@@ -79,6 +130,24 @@ definitions:
           jcr:uuid: 7ab31fb7-1720-445c-9ca3-0ff2ce4e11ab
           hippostdpubwf:lastModificationDate: 2021-05-18T11:53:12.350+01:00
           hippostdpubwf:creationDate: 2021-05-18T11:53:12.350+01:00
+        /hipposysedit:prototype[2]:
```
Looks like another duplicate node - could we remove if its redundant ?

Raq-HEE replied:
Removed

smanivasag commented on repository-data/application/src/main/resources/hcm-content/content/documents/administration/valuelists/documenttypes.yaml:
```diff
@@ -31,3 +31,15 @@
       jcr:primaryType: selection:listitem
       selection:key: hee:blogPost
       selection:label: BlogPost
+    /selection:listitem[6]:
```
Could we reload this value-list via `hcm-actions.yaml` ?

Raq-HEE replied:
Added

smanivasag commented on repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank.yaml:
```diff
@@ -122,3 +122,121 @@
         jcr:encoding: UTF-8
         jcr:lastModified: 2021-05-11T12:32:18.067+01:00
         jcr:mimeType: application/vnd.openxmlformats-officedocument.wordprocessingml.document
+  /search-search-bank:
```
Could we reload `/content/documents/lks/search-bank` ?

Raq-HEE replied:
Added

smanivasag commented on repository-data/development/src/main/resources/hcm-content/content/documents/lks/listing-pages.yaml:
```diff
@@ -78,13 +78,13 @@
     jcr:mixinTypes: ['hippo:named', 'hippo:versionInfo', 'mix:referenceable']
```
Could we reload `/content/documents/lks/listing-pages` ?

Raq-HEE replied:
Added.

smanivasag commented on repository-data/development/src/main/resources/hcm-content/content/documents/lks/events.yaml:
```diff
@@ -262,3 +262,64 @@
       hippostdpubwf:publicationDate: 2021-05-18T12:01:49.739+01:00
       hippotranslation:id: 300dfc60-c232-4d9d-8a91-ef14de917802
       hippotranslation:locale: en
+  /search-event:
```
Could we reload `/content/documents/lks/events` ?

Raq-HEE replied:
Added. I've added a few others also.

smanivasag commented on repository-data/development/src/main/resources/hcm-content/content/documents/lks/case-studies.yaml:
```diff
@@ -111,3 +111,91 @@
           jcr:encoding: UTF-8
           jcr:lastModified: 2021-05-11T13:06:13.939+01:00
           jcr:mimeType: application/pdf
+  /search-case-study:
```
Could we reload `/content/documents/lks/case-studies` ?

Raq-HEE replied:
Added

smanivasag commented on repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/list-item.ftl:
```diff
@@ -203,7 +203,7 @@
 
         <#if ['Bulletin', 'CaseStudy', 'SearchBank', 'Event']?seq_contains(item.class.simpleName) || (pageURL != pageNotFoundURL || ('uk.nhs.hee.web.beans.Guidance' == item.getClass().getName() && miniHubGuidancePathToURLMap[item.path]??))>
```
The URL for `Bulletin [probably an existing bug], CaseStudy, SearchBank & Event` ends up with 404. Could we fix them ?

Raq-HEE replied:
Thanks for spotting this. The links are now resolving to the relevant urls/binaries. 

I've not displayed exactly as the listing pages for each doc type, something I thought may be worth discussing with design team before we refactor the entire code pice for this as we probably can create macros to be shared with the listItem code blocks above.

smanivasag replied:
Thanks @Raq-HEE - Moved `<#case "uk.nhs.hee.web.beans.Guidance">` case on `list-item.ftl` right above `default` case so as to ensure rendering non-minihub guidance pages (using `default` case). Have done this work as part of 29d75f73c74148f08988df7a3638153f96ca1aa9 commit. Could you take a look ? Thanks!

smanivasag commented on repository-data/webfiles/src/main/resources/site/freemarker/hee/catalog/searchresults-main.ftl:
```diff
@@ -39,24 +39,24 @@
-                        <form class="nhsuk-filter" method="get" action="${pagelink}">
-                            <@fmt.message key="filters.label" var="filtersLabel"/>
-                            <p class="nhsuk-filter__title nhsuk-heading-l">${filtersLabel}</p>
+<#--                    <div class="nhsuk-grid-column-one-third">-->
```
If it's OK, could we change these inline comments to one block comment so as to avoid FTL engine to process too many comments ?

Raq-HEE replied:
Done

smanivasag replied:
Thanks @Raq-HEE - Reverted `ftlvariables` comment updates (as it would be good to keep them as inline comments for easy read) and separated out `contentTypesMap` & `selectedContentTypes` variables for future use if required. Have done this work as part of 29d75f7 commit. Hope this is OK ? Thanks!

smanivasag commented on repository-data/webfiles/src/main/resources/site/freemarker/hee/catalog/searchresults-main.ftl:
```diff
@@ -39,24 +39,24 @@
             </form>
```
Could we comment the following active filters section & contentTypes hidden element as well ?
```
<#-- Active Filters -->
<#if selectedContentTypes?has_content>
    <div class="nhsuk-listing__active-filters nhsuk-u-margin-bottom-5">
        <#list selectedContentTypes as contentType>
            <div class="nhsuk-filter-tag nhsuk-tag" data-filter="${contentType}">
                <span>${contentTypesMap[contentType]}</span>
                <@hst.link path='/static/assets/icons/icon-close-white.svg' var="closeIcon"/>
                <img class="nhsuk-filter-tag__icon" src="${closeIcon}" alt="Remove" hidden/>
            </div>
        </#list>
    </div>
</#if>
<#-- End Active Filters -->
```

```
<#list selectedContentTypes as contentType>
    <input type="hidden" name="contentTypes" value="${contentType}">
</#list>
```

Raq-HEE replied:
Done

smanivasag commented on repository-data/webfiles/src/main/resources/site/freemarker/hee/catalog/searchresults-main.ftl:
```diff
@@ -39,24 +39,24 @@
             </form>
             <div class="nhsuk-listing">
```
I think we may need to comment or refactor `uk.nhs.hee.web.components.SearchResultsComponent` class to not to use `contentTypesMap` & `selectedContentTypes` and it's references.

Raq-HEE replied:
Done. Commented for now.

smanivasag reviewed (commented):
Hi @Raq-HEE - Thanks for the PR 👍

Raq-HEE commented:
Hi @smanivasag - thanks for reviewing this. I've addressed most of your comments. A couple of areas for contention are:
- list-item.ftl
  - the entire code probably needs refactoring if we're to display these like the listing pages to use common macros. thought it may be best for design team to review what we have currently before further effort spent on this. 
- repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[1]/searchDocument/hippo-resource
  - I wasn't sure if the extra nodes are duplicates and need to be removed. Also wether we can safely remove hippo-resource files?
- Whether we ought to remove the comments, as we can always refer back to a previous commit, if we need to re-introduce.

smanivasag commented:
> Hi @smanivasag - thanks for reviewing this. I've addressed most of your comments. A couple of areas for contention are:
> 
> * list-item.ftl
>   
>   * the entire code probably needs refactoring if we're to display these like the listing pages to use common macros. thought it may be best for design team to review what we have currently before further effort spent on this.
> * repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[1]/searchDocument/hippo-resource
>   
>   * I wasn't sure if the extra nodes are duplicates and need to be removed. Also wether we can safely remove hippo-resource files?
> * Whether we ought to remove the comments, as we can always refer back to a previous commit, if we need to re-introduce.

Hi @Raq-HEE - Thanks for addressing all the review comments. Have made 29d75f73c74148f08988df7a3638153f96ca1aa9 commit containing minor fixes. Could you review ?

- Re. `list-item.ftl` refactoring - Sure, I am happy for design team to feedback on this.

- Re. `hippo-resource` - `hippo-resource` are seem to be empty placeholder files that brXM creates when the author doesn't add a file for the document field i.e. It seems to be default file for empty document fields. I think the following files can safely be removed as they are not being referenced in `repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank.yaml`:
```
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[1]/searchDocument/hippo-resource
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[2]/searchDocument/hippo-resource
```

But, I think we still need the following files as they are being referenced in `repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank.yaml`:
```
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[1]/strategyDocument/hippo-resource
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[2]/strategyDocument/hippo-resource
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[3]/strategyDocument/hippo-resource
```
Re. commenting out the filter functionality - I would probably lean towards removing the commented out filter functionality snippets as we can always refer back the previous commit if required as you mentioned. A new team member may not know that the filter functionality existed previously, but this doesn't qualifies to keep the commented out snippets. Thanks!

smanivasag commented:
### Update
Re. `list-item.ftl` refactoring - The PR will be deployed to dev for design team review (as agreed with Robin).

Re. hippo-resource - Removed the following unreferenced `hippo-resource` files:
```
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[1]/searchDocument/hippo-resource
repository-data/development/src/main/resources/hcm-content/content/documents/lks/search-bank/search-search-bank/search-search-bank[2]/searchDocument/hippo-resource
```
Have committed this work as part of e38e7f6de5d2661022fb9335491f41b182c64352.

Re. commenting out the filter functionality - Removed the `commented out` search filter functionality code snippets from freemarker templates as well as from component java class as this functionality could be referenced back via previous commits later if required. Have committed this work as part of 0a06f7ebec9c14884820c1a53e1a306256b7765f.
